tests/test_builtin_traits.py

```diff
@@ -160,3 +165,40 @@ def test_is_terminator_fails_if_not_last_operation_parent_block():
         VerifyException, match="must be the last operation in the parent block"
     ):
         op0.verify()
+
+
+@irdl_op_definition
+class IsolatedFromAboveOp(IRDLOperation):
+    """
+    An isolated from above operation.
+    """
+
+    name = "test.isolated_from_above"
+
+    region: Region = region_def()
+
+    traits = frozenset([IsolatedFromAbove()])
+
+
+def test_isolated_from_above():
+    op = IsolatedFromAboveOp(regions=[Region()])
+    op.verify()
+
+    block = Block(arg_types=[builtin.i32])
+    block.add_op(arith.Addi(block.args[0], block.args[0]))
+
+    op = IsolatedFromAboveOp(regions=[Region([block])])
+    op.verify
+
+    out_cst = arith.Constant.from_int_and_width(0, builtin.i32)
+    out_block = Block(
```

Comment:
maybe also a test case with 2 nested IsolatedFromAbove ops?

Reply:
Yeah, I was lazy on that one :sweat_smile: Mostly lazy on thinking about the exact examples I wanted; please give ideas if you have others :wink:

Reply:
yeah, no worries, I was just thinking of exercising that last condition (`if not child_op.has_trait(IsolatedFromAbove):`.